src/consumer.ts

```diff
@@ -164,6 +165,24 @@ export class Consumer extends EventEmitter {
     return !this.stopped;
   }
 
+  /**
+   * Returns true if polling has occurred within the expected
+   * polling wait time interval (`pollingWaitTimeMs`).
+   * @note - Contrast with `isRunning`, which indicates whether
+   * polling SHOULD be happening. `isHealthy` indicates whether
+   * there is evidence polling is actually happening.
+   */
+  public get isHealthy(): boolean {
+    if (!this.lastPollAt && !this.stopped) {
+      return false;
+    }
+    if (!this.lastPollAt) {
+      return true;
+    }
+    const timeElapsedSinceLastPoll = this.lastPollAt - Date.now();
+    return this.pollingWaitTimeMs > timeElapsedSinceLastPoll;
```

Comment:
Hmm, I'm not sure if this will work. What I mean:
```js
let pollingWaitTimeMs = 5000;

// poll
let lastPollAt = Date.now(); // 1656400864961

// isHealthy()
pollingWaitTimeMs > lastPollAt - Date.now(); // 5000 > 1656400864961 - [whatever higher then lastPollAt]
// => Always true
```
`isHealthy` is always true as `lastPollAt - Date.now()` is equal to `(-infinity, 0]` interval.

Reply:
Ah yes. Makes sense. I've fixed the logic. I realized as well that my tests always passed. I've fixed them so that they will fail with the incorrect previous logic.

Reply:
Now it gets to be unhealthy for a split second every time before polling. I would change the `>` to `>=`, as a new poll is triggered not before but exactly on the `pollingWaitTimeMs`. And would add some buffer to the `pollingWaitTimeMs`, as the `lastPollAt` receives an updated timestamp a few micro/milliseconds after the start of polling.
```js
let buffer = 100; // ms
return this.pollingWaitTimeMs + buffer >= timeElapsedSinceLastPoll;
```

Reply:
I think we should only consider the base case when `setTimeout` is functioning normally. It is possible for the timer to be [disrupted by other processes on the machine](https://abhi9bakshi.medium.com/why-javascript-timer-is-unreliable-and-how-can-you-fix-it-9ff5e6d34ee0), but I think it's beyond the scope of this code change.

For example, if we knew that we were performing some heavy operations on our machine, we could adjust the K8S probe to have a `failureThreshold` of more than 1, which would give us the buffer you mention.

I've changed the comparison to `>=` though I could not simulate a case in my tests where they were every actually equal.